python/pyspark/ml/param/__init__.py

```diff
@@ -39,6 +39,15 @@ def __init__(self, parent, name, doc):
         self.name = str(name)
         self.doc = str(doc)
 
+    def _copy_new_parent(self, parent):
```

Comment:
(To be clear I'm open to another name but maybe I'll wait to swap it for @jkbradley or @JoshRosen  or @mengxr) to also way in on it.